src/FinfoMimeTypeDetector.php

```diff
@@ -53,9 +53,14 @@ public function __construct(
 
     public function detectMimeType(string $path, $contents): ?string
     {
-        $mimeType = is_string($contents)
-            ? (@$this->finfo->buffer($this->takeSample($contents)) ?: null)
-            : null;
+        $mimeType = null;
+        if (is_string($contents)) {
+            $mimeType = @$this->finfo->buffer($this->takeSample($contents));
+        } elseif (is_resource($contents) && get_resource_type($contents) === 'stream') {
```

Comment:
The `if` condition should include a check to see if the stream is seekable, otherwise the calls below result in an error.

Reply:
Addressed by generally ignoring non-seekable streams, while providing an option to change this behavior when creating the class.